tests/test_crystal_ball.py

```diff
@@ -1,8 +1,8 @@
 """Module for testing the htcrystalball module."""
 from htcrystalball import examine, collect, utils
 from os import path
+from pytest import fixture, raises as praises
```

Comment:
I've seen this as a trend among python devs, so I assume it's best practice to limit the imports. But I've long enjoyed the namespacing of importing the entire module (easier to trace that way). Can you explain (or point to a discussion explaining) the benefits of reducing the scope of imports within a module?

Reply:
There is no best practices here. Importing the entire package (vs. importing modules) is the same in terms of speed or what gets imported. The package is always imported in its entirety (although I'm sure there are exceptions, for example with packages that don't include all of their contents in the init file). For a big script I'd recommend going for `import pytest` and using `@pytest.fixture` and `pytest.raises` for readability and self documentation. For a small script such as this, especially if it is built around pytest, it shows precisely what functions from pytest we're using. In hindsight, I think `raises as praises` is not necessary (although I do appreciate the pun). It is simply a matter of style. For `from htcrystalball import examine, collect, utils` it confers a bigger benefit. We now know every module that is being tested.

A side-benefit is brevity in lines. While certainly not important (or something that necessitates these changes), `examine` is shorter than `htcrystalball.examine`, which can prevent awkward line-breaks at the 70 character limit. I would not recommend this for a big script, but for a small script like this the overview is not lost.